olp-cpp-sdk-core/tests/thread/ContinuationTest.cpp

```diff
@@ -53,145 +56,167 @@ class ContinuationTest : public testing::Test {
 };
 
 TEST_F(ContinuationTest, MultipleThen) {
-  std::promise<int> promise;
+  std::promise<ResponseType<int>> promise;
   auto future = promise.get_future();
 
-  auto continuation = Create([](olp::thread::ExecutionContext,
-                                std::function<void(int)> next) { next(1); })
-                          .Then([](olp::thread::ExecutionContext, int,
-                                   std::function<void(int)> next) { next(2); })
-                          .Finally([&promise](Response<int> response) {
-                            promise.set_value(response.GetResult());
-                          });
+  int counter = 0;
+  auto continuation =
+      Create([&](olp::thread::ExecutionContext, std::function<void(int)> next) {
+        next(++counter);
+      })
+          .Then([&](olp::thread::ExecutionContext, int,
+                    std::function<void(int)> next) { next(++counter); })
+          .Finally([&promise](ResponseType<int> response) {
+            promise.set_value(std::move(response));
+          });
   continuation.Run();
 
-  EXPECT_EQ(future.get(), 2);
+  ASSERT_EQ(future.wait_for(kMaxWaitMs), std::future_status::ready);
+  auto result = future.get();
+
+  ASSERT_EQ(result.GetResult(), counter);
```

Comment:
You also need to check if the response is successful before getting the result.

The last check in a test can be EXPECT, instead of ASSERT.